src/sim/UI/ViewClass.py

```diff
@@ -186,6 +198,45 @@ def __init__(self, target, args):
 
         super().__init__()
 
+
     @pyqtSlot()
     def run(self):
-        self.__target(self.__args)
\ No newline at end of file
+        self.__target(self.__args)
+
+
+# Custom QGraphicsEllipseItem
+class NamedEllipseItem(QGraphicsEllipseItem):
+    def __init__(self, name, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.__name = name
+
+
+    def getName(self):
+        return self.__name
+    
+
+# TODO: I don't like having to pass the controller to this class, need to update to find a work around for that
+class CustomQGraphicsScene(QGraphicsScene):
+    def __init__(self, controller, parent=None):
+        super().__init__(parent)
+        self.__controller = controller
+
+    def mouseReleaseEvent(self, event):
+        # Call the base class method to handle normal mouse press events
+        super().mouseReleaseEvent(event)
+        
+        # Custom release handler TODO: UPDATE TO MOVE PROCESSING TO WORKER
+        self.__CustomReleaseHandler(event)
+
+
+    # Custom release handler to call controller and update robot positions
+    def __CustomReleaseHandler(self, event):
+        if event.button() == Qt.LeftButton:
+            mousePos = event.scenePos()
+            items = self.items(mousePos)
+            for item in items:
+                # Need check because items contains both Ellipse & Text but Ellipse is the only one with updated coordinates
```

Comment:
text coordinates should be updated, but i havent checked that. not sure why that wouldn't be the case

Reply:
its probably better to use the ellipse coords, anyway